modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,390 @@
+                                            c <<= 2;
+                                        }
+                                    }
+                                    if (ii != i || jj != j)
+                                    {
+                                        if (image2[rrWidth + jj]  > image2[rWidth + j] - t)
+                                        {
+                                            c2 <<= 2;
+                                            c2 |= 0x3;
+                                        }
+                                        else if (image2[rWidth + j] - t < image2[rrWidth + jj] && image2[rWidth + j] + t >= image2[rrWidth + jj])
+                                        {
+                                            c2 <<= 2;
+                                            c2 = c2 + 1;
+                                        }
+                                        else
+                                        {
+                                            c2 <<= 2;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        else if (type == MCT::MeanVariation)
+                        {
+                            //to do mean variation
+                        }
+                        dst1[distV + j] = c;
+                        dst2[distV + j] = c2;
+                    }
+                }
+            }
+
+        };
+        //!paralel implementation of the center symetric census
+        class parallelSymetricCensus:public ParallelLoopBody
+        {
+        private:
+            uint8_t *image1, *image2;
+            int *dst1, *dst2;
+            int n2, width, height, type;
+        public:
+            parallelSymetricCensus(uint8_t * img1, uint8_t * img2, int w, int h, int k2, int * distance1, int * distance2,const int t) :
+                image1(img1), image2(img2), dst1(distance1), dst2(distance2), n2(k2), width(w), height(h), type(t){}
+
+            virtual void operator()(const cv::Range &r) const {
+
+                for (int i = r.start; i <= r.end ; i++)
+                {
+                    int distV = (i)* width;
+                    for (int j = n2; j <= width - n2; j++)
+                    {
+                        int c = 0;
+                        int c2 = 0;
+                        //the classic center symetric census which compares the curent pixel with its symetric not its center.
+                        if (type == SymetricCensus::ClassicCenterSymetricCensus)
+                        {
+                            for (int ii = -n2; ii < 0; ii++)
+                            {
+                                int rrWidth = (ii + i) * width;
+                                for (int jj = -n2; jj <= +n2; jj++)
+                                {
+                                    if (ii != i || jj != j)
```

Comment:
I still don't get it. 

- `ii` is in the range `[-n2, 0)`, i.e. always negative. Agree?
- `i` is in the range `[n2, width - n2)`, i.e. always positive. Agree?
- `ii` is always negative, `i` is always positive, i.e. `ii != i` is always true. Agree?